core/src/main/scala/cats/instances/eq.scala

```diff
@@ -22,8 +22,45 @@
 package cats
 package instances
 
-trait EqInstances extends kernel.instances.EqInstances {
-  implicit val catsContravariantMonoidalForEq: ContravariantMonoidal[Eq] =
+trait EqInstances extends kernel.instances.EqInstances with EqInstances0 {
+  implicit def catsDecidableForEq: Decidable[Eq] =
+    new Decidable[Eq] {
+
+      /**
+       * Defaults to the trivial equivalence relation
+       * contracting the type to a point
+       */
+      val unit: Eq[Unit] = Eq.allEqual
+
+      /**
+       * Derive an `Eq` for `B` given an `Eq[A]` and a function `B => A`.
+       *
+       * Note: resulting instances are law-abiding only when the functions used are injective (represent a one-to-one mapping)
+       */
+      def contramap[A, B](fa: Eq[A])(f: B => A): Eq[B] =
+        Eq.by(f)(fa)
+
+      def product[A, B](fa: Eq[A], fb: Eq[B]): Eq[(A, B)] =
+        Eq.instance { (left, right) =>
+          fa.eqv(left._1, right._1) && fb.eqv(left._2, right._2)
+        }
+
+      def sum[A, B](fa: Eq[A], fb: Eq[B]): Eq[Either[A, B]] =
+        Eq.instance {
+          case (Left(a1), Left(a2))   => fa.eqv(a1, a2)
+          case (Right(b1), Right(b2)) => fb.eqv(b1, b2)
+          case _                      => false
+        }
+
+      override val zero: Eq[Nothing] = Eq.allEqual[Nothing]
+    }
+
+  val catsContravariantMonoidalForEq: ContravariantMonoidal[Eq] =
+    catsContravariantMonoidalForEqForBinCompat
+}
+
+private[instances] trait EqInstances0 {
+  implicit def catsContravariantMonoidalForEqForBinCompat: ContravariantMonoidal[Eq] =
```

Comment:
Actually ... I don't think we need this at all 😅 this is completely subsumed by the new `Decidable` instance :)